realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -124,4 +136,137 @@ protected static Realm getRealm(RealmObject obj) {
     protected static Row getRow(RealmObject obj) {
         return obj.row;
     }
+
+    /**
+     * Set the Future instance returned by the worker thread, we need this instance
+     * to force {@link #load()} an async query, we use it to determine if the current
+     * RealmResults is a sync or async one
+     *
+     * @param pendingQuery pending query
+     */
+    void setPendingQuery(Future<Long> pendingQuery) {
+        this.pendingQuery = pendingQuery;
+        if (isLoaded()) {
+            // the query completed before RealmQuery
+            // had a chance to call setPendingQuery to register the pendingQuery (used btw
+            // to determine isLoaded behaviour)
+            onCompleted();
+
+        } // else, it will be handled by the Realm#handler
+    }
+
+    /**
+     * Set the class type of this RealmObject, we need it to figure out the {@link Table}
+     *
+     * @param clazz type/table
+     */
+    void setType(Class<? extends RealmObject> clazz) {
```

Comment:
This seems a bit iffy just to be able to realm.getTable(). The dynamic API has similar requirements so I think there are better ways of doing this.